spectroscopy/code_src/Euclid_functions.py

```diff
@@ -0,0 +1,185 @@
+import astropy.units as u
+import numpy as np
+import pandas as pd
+from astropy.coordinates import SkyCoord
+from astropy.table import QTable
+from astropy.io import fits
+from astroquery.ipac.irsa import Irsa
+
+from data_structures_spec import MultiIndexDFObject
+import warnings
+from astropy.utils.exceptions import AstropyWarning
+
+warnings.simplefilter('ignore', category=AstropyWarning)
+
+
+def get_coord_from_objectid(object_id, table_mer="euclid_q1_mer_catalogue"):
+    """
+    Query IRSA's cloud-based TAP service to retrieve the sky coordinates of a Euclid object.
+    This function performs an ADQL query against the specified MER catalog hosted at IRSA
+    and returns the right ascension and declination of the object as a SkyCoord instance.
+
+    Parameters
+    ----------
+    object_id : int or str
+        The Euclid object ID to look up.
+
+    table_mer : str, optional
+        The name of the IRSA TAP-accessible MER catalog table to query.
+        Defaults to "euclid_q1_mer_catalogue".
+
+    Returns
+    -------
+    coord : astropy.coordinates.SkyCoord or None
+        The sky coordinates (RA, Dec) of the object, if found.
+        Returns None if the query fails or if no match is found.
+
+    Notes
+    -----
+    - This function queries a remote (cloud-based) TAP service and requires internet access.
+    - If no result is found or if the query fails, a warning is printed and None is returned.
+    """
+    
+    adql_query = f"""
+    SELECT ra, dec FROM {table_mer}
+    WHERE object_id = {object_id}
+    
+    """
+    try:
+        result = Irsa.query_tap(adql_query).to_table()
+        if len(result) == 0:
+            print(f"No match found for object_id {object_id}")
+            return None
+        ra, dec = result[0]["ra"], result[0]["dec"]
+        return SkyCoord(ra=ra * u.deg, dec=dec * u.deg)
+    except Exception as e:
+        print(f"Query failed for object_id {object_id}: {e}")
+        return None
+
+
+def Euclid_get_spec(sample_table, search_radius_arcsec):
```

Comment:
snake case should be all lower case, no matter if it has a proper noun

```suggestion
def euclid_get_spec(sample_table, search_radius_arcsec):
```